src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.ClientCertProvider.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+
+using CURLoption = Interop.libcurl.CURLoption;
+using CURLcode  = Interop.libcurl.CURLcode;
+using CURLINFO = Interop.libcurl.CURLINFO;
+using SslClientCertCb = Interop.libcurl.client_cert_cb;
+using SafeSslContextHandle = Interop.libssl.SafeSslContextHandle;
+
+namespace System.Net.Http
+{
+    internal partial class CurlHandler : HttpMessageHandler
+    {
+        [Conditional(VerboseDebuggingConditional)]
+        private static void IgnorePeerCert(EasyRequest easy)
+        {
+            easy.SetCurlOption(CURLoption.CURLOPT_VERIFYPEER, 0L);
```

Comment:
I assumed VerboseDebugging is not for production use.

This option is at libcurl end means that we do not verify the server's certificate. This does not impact the current feature's functionality. As there's no public API at HttpClientHandler level that allows us to ignore server certificates, I used this approach. Setting VERIFY_PEER to 0 helps in testing the httpClient against different servers with different certs - even local IIS servers.   And this is being set only when the clientcertificateoptions is set to automatic.